tests/Fixer/ClassNotation/OrderedTypesFixerTest.php

```diff
@@ -0,0 +1,626 @@
+     *
+     * @requires PHP 8.1
+     */
+    public function testFixPhp81(string $expected, ?string $input = null, ?array $config = null): void
+    {
+        if (null !== $config) {
+            $this->fixer->configure($config);
+        }
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return iterable<string[]|(string|null|array<string, string>)[]>
+     */
+    public static function providePhp81Cases(): iterable
+    {
+        yield [
+            "<?php\nclass Foo\n{\n    public A&B \$bar;\n}\n",
+            "<?php\nclass Foo\n{\n    public B&A \$bar;\n}\n",
+        ];
+
+        yield [
+            "<?php\nclass Foo\n{\n    public Ae&At \$bar;\n}\n",
+            "<?php\nclass Foo\n{\n    public At&Ae \$bar;\n}\n",
+        ];
+
+        yield [
+            "<?php\nclass Foo\n{\n    public readonly null|A|B \$bar;\n}\n",
+            "<?php\nclass Foo\n{\n    public readonly B|null|A \$bar;\n}\n",
+        ];
+
+        yield [
+            "<?php\nclass Foo\n{\n    public readonly A|B|null \$bar;\n}\n",
+            "<?php\nclass Foo\n{\n    public readonly B|null|A \$bar;\n}\n",
+            ['null_adjustment' => 'always_last'],
+        ];
+
+        yield [
+            "<?php\nclass Foo\n{\n    public readonly A|null|X \$bar;\n}\n",
+            "<?php\nclass Foo\n{\n    public readonly X|A|null \$bar;\n}\n",
+            ['null_adjustment' => 'none'],
+        ];
+
+        yield [
+            "<?php\nclass Foo\n{\n    public B&A \$bar;\n}\n",
+            null,
+            ['sort_algorithm' => 'none'],
+        ];
+    }
+
+    /**
+     * Provisional support for PHP 8.2's Disjunctive Normal Form (DNF) Types.
+     *
+     * @dataProvider providePhp82Cases
+     *
+     * @param null|array<string, string> $config
+     *
+     * @requires PHP 8.2
+     */
+    public function testFixPhp82(string $expected, ?string $input = null, ?array $config = null): void
+    {
+        if (null !== $config) {
+            $this->fixer->configure($config);
+        }
+
+        $this->doTest($expected, $input);
+    }
+
+    /**
+     * @return iterable<string[]|(string|null|array<string, string>)[]>
+     */
+    public static function providePhp82Cases(): iterable
+    {
+        yield [
+            "<?php\nclass Foo\n{\n    public null|array|(At&Bz)|string \$bar = null;\n}\n",
+            "<?php\nclass Foo\n{\n    public string|(Bz&At)|array|null \$bar = null;\n}\n",
+        ];
+
+        yield [
+            "<?php\nclass Foo\n{\n    public array|(At&Bz)|string|null \$bar = null;\n}\n",
+            "<?php\nclass Foo\n{\n    public string|(Bz&At)|array|null \$bar = null;\n}\n",
+            ['null_adjustment' => 'always_last'],
+        ];
+
+        yield [
+            "<?php\nclass Foo\n{\n    public array|(At&Bz)|null|string \$bar = null;\n}\n",
+            "<?php\nclass Foo\n{\n    public string|(Bz&At)|array|null \$bar = null;\n}\n",
+            ['null_adjustment' => 'none'],
+        ];
```

Comment:
Can we have a case like `(D&B)|(C&A)` (and more) with alpha sorting to see if the type won't change the group?

Reply:
Did I address your concern correctly in https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/commit/ebb1825f63cfbefc72b9897821c4e2d1666db3b5?

Reply:
Yes, more than enough, thank you